python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -57,61 +61,91 @@ def get(self, mine_no_or_guid=None):
                 result = list((map(lambda x: x.json_for_map(), records)))
                 return {'mines': result}
 
-            # Handle ListView request
-            items_per_page = request.args.get('per_page', 25, type=int)
-            page = request.args.get('page', 1, type=int)
-            search_term = request.args.get('search', None, type=str)
-            status_search_term = request.args.get('status', None, type=str)
-
-            # Create a filter on mine status if one is provided
-            if status_search_term:
-                status_search_term_array = status_search_term.split(',')
-                status_filter = MineStatusXref.mine_operation_status_code.in_(
-                    status_search_term_array)
-                status_reason_filter = MineStatusXref.mine_operation_status_reason_code.in_(
-                    status_search_term_array)
-                status_subreason_filter = MineStatusXref.mine_operation_status_sub_reason_code.in_(
-                    status_search_term_array)
-                all_status_filter = status_filter | status_reason_filter | status_subreason_filter
+            return self.mine_filter_and_search_method(request.args)
 
-            if search_term:
-                name_filter = Mine.mine_name.ilike('%{}%'.format(search_term))
-                number_filter = Mine.mine_no.ilike('%{}%'.format(search_term))
-                permit_filter = Permit.permit_no.ilike('%{}%'.format(search_term))
-                mines_query = Mine.query.filter(name_filter | number_filter)
-                permit_query = Mine.query.join(Permit).filter(permit_filter)
-                mines_permit_join_query = mines_query.union(permit_query)
-                if status_search_term:
-                    status_query = Mine.query\
-                        .join(MineStatus)\
-                        .join(MineStatusXref)\
-                        .filter(all_status_filter)
-                    mines_permit_join_query = mines_permit_join_query.intersect(status_query)
-                result_query, pagination_details = apply_pagination(mines_permit_join_query, page,
-                                                                    items_per_page)
-
-            else:
-                sort_criteria = [{'model': 'Mine', 'field': 'mine_name', 'direction': 'asc'}]
-                if status_search_term:
-                    mine_query_with_status = Mine.query\
-                        .join(MineStatus)\
-                        .join(MineStatusXref)\
-                        .filter(all_status_filter)
-                    result_query = apply_sort(mine_query_with_status, sort_criteria)
-                else:
-                    result_query = apply_sort(Mine.query, sort_criteria)
-
-            paginated_mine_query, pagination_details = apply_pagination(
-                result_query, page, items_per_page)
+    def mine_filter_and_search_method(self, args):
```

Comment:
Thanks for breaking this out! A couple things:
1) `_method` is redundant here
2) The `get` method should still return the JSON payload
3) Personal opinion: either use a verb method name (like `apply_search_filters`) or noun method name (like `filtered_results`)

You should be able to do something like this to separate these concerns:
```python
# inside of the get method:
mines, pagination_details = mine_filter_and_search(request.args)
return {
  'mines': list(map(lambda x: x.json_for_list(), mines)),
  ...
}
```